Madness/Parser.swift

```diff
@@ -106,8 +106,7 @@ private func memoize<T>(f: () -> T) -> () -> T {
 }
 
 public func delay<C: CollectionType, T>(parser: () -> Parser<C, T>.Function) -> Parser<C, T>.Function {
-	let memoized = memoize(parser)
-	return { memoized()($0, $1) }
+	return { memoize(parser)()($0, $1) }
```

Comment:
I think this may be a performance regression, and may even be incorrect.

By removing the temporary value, the memoized parser is in fact not memoized at all—it gets recreated every time the `delay`ed parser is evaluated.